homeassistant/components/modbus/binary_sensor.py

```diff
@@ -112,8 +112,10 @@ async def async_update(self, now: datetime | None = None) -> None:
             self._attr_available = True
             self._result = result
             if self._input_type in (CALL_TYPE_COIL, CALL_TYPE_DISCRETE):
+                self._result = result.bits
                 self._attr_is_on = bool(result.bits[0] & 1)
             else:
+                self._result = result.registers
                 self._attr_is_on = bool(result.registers[0] & 1)
```

Comment:
These calculations does not match how it's done done the callback below?

Reply:
It is actually the same, let me explain:
- the primary sensor gets one of 2 arrays (bits/registers), and sets self._result which is sent to the coordinator, so that the slaves only have 1 array.
- the primary sensor always uses registers/bits index 0, whereas slaves uses index 1-n.
- the calculation itself is identical: bool(<value> & 1)

I hope that explains things.

Reply:
Move the is_on calculation outside if the if clause and use self._result as variable instead, then code will be identical in both cases and match slave sensor.